spec/order_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'order'
+
+describe Order do
+  it { is_expected.to respond_to(:basket) }
+  it { is_expected.to respond_to(:menu) }
+  it { is_expected.to respond_to(:add).with(2).argument }
+  it { is_expected.to respond_to(:review_order) }
+  it { is_expected.to respond_to(:place_order) }
+
+  describe "#basket" do
+    it "should be an Array" do
+      expect(subject.basket).to be_a Array
+    end
+
+    it "should be empty upon initialize" do
+      expect(subject.basket).to be_empty
+    end
+
+  end
+
+  describe "#menu" do
+    it "should be an instance of Menu" do
+      expect(subject.menu).to be_an_instance_of(Menu)
+    end
+  end
+
+  describe "#add" do
+    context "selecting item from menu" do
+
+      before do
+        subject.add("pancakes")
+      end
+
+      it "should add the item to the basket" do
+        expect(subject.basket).to include("pancakes" => 5)
```

Comment:
It feels like this is testing the state of `Order`, rather than its behaviour.  This means that if the implementation of basket changes, this test might need to be updated for the new implementation.  Is there a way to better encapsulate `basket` and test the behaviour of `Order` to avoid this?